server/src/main/java/org/elasticsearch/index/mapper/GeoPointScriptFieldType.java

```diff
@@ -107,13 +102,8 @@ public Query existsQuery(SearchExecutionContext context) {
     }
 
     @Override
-    public Query geoShapeQuery(Geometry shape, String fieldName, ShapeRelation relation, SearchExecutionContext context) {
-        final LatLonGeometry[] luceneGeometries = GeoShapeUtils.toLuceneGeometry(fieldName, context, shape, relation);
-        if (luceneGeometries.length == 0
-            || (relation == ShapeRelation.CONTAINS && Arrays.stream(luceneGeometries).anyMatch(g -> (g instanceof Point) == false))) {
-            return new MatchNoDocsQuery();
-        }
-        return new GeoPointScriptFieldGeoShapeQuery(script, leafFactory(context), fieldName, relation, luceneGeometries);
+    public Query geoShapeQuery(SearchExecutionContext context, String fieldName, ShapeRelation relation, LatLonGeometry... geometries) {
+        return new GeoPointScriptFieldGeoShapeQuery(script, leafFactory(context), fieldName, relation, geometries);
```

Comment:
Are we loosing an optimisation here by deleting the check for any geometries not being Points when we have CONTAINS?

Reply:
Or perhaps this is fixing a bug, because the previous condition of 'return no results if the relation is CONTAINS and there are any non-points in the search' seems a bit unreasonable to me.

Reply:
I recover the optimization. the only way for an indexed point to contain a query shape is that the query shape is a point or multi-point?